### ISSUE TITLE
Content Editor cannot create link in WYSIWYG editor (nor anchor / email / file)

### ISSUE THREAD

**Issue by baukezwaan** [silverstripe/silverstripe-subsites#357]
## Affected Version

```
phpunit/phpunit            5.7.27             The PHP Unit Testing framework.
silverstripe-themes/simple dev-master 4d546a4 The SilverStripe simple theme (default SilverStripe 3 theme)
silverstripe/recipe-cms    1.0.3              SilverStripe recipe for fully featured page and asset content editing
silverstripe/recipe-plugin 1.2.0              Helper plugin to install SilverStripe recipes
silverstripe/subsites      2.0.1              Run multiple sites from a single SilverStripe install.
```

Issue appears to be on SilverStripe CMS version 4.0.3 and 4.1

## Description

A user that belongs to a group without 'Access to all CMS sections' is getting an error like
`Unexpected token < in JSON at position 0`
when you try to insert a link or anchor into the HTMLeditor on a page.

![knipsel](https://user-images.githubusercontent.com/795238/38736820-91b4753e-3f2d-11e8-9dc4-325c71056923.PNG)

## Steps to Reproduce

On a clean SilverStripe installation with Subsites, create a user that belongs to the Content Author group. Then try to insert a link into a page.

**Comment by baukezwaan** [comment]
A little bit of experimenting with xdebug seems to point to LeftAndMainSubsites->canAccess. 

It checks to see if the current Member has ADMIN or CMS_ACCESS_LeftAndMain. My Member doesn't have these rights, but have only access to certain parts of the CMS. 
Additionally, canAccess check if we have access to current section on the current Subsite, but that doesn't seem to account for the possibility that we are trying to access an action on LeftAndMain directly. 

I don't know what would be a proper fix, but when I insert an if( $oController instanceof LeftAndMain ) return true; I at least can access the TinyMCE modals.

**Comment by adrexia** [comment]
This appears to be fixed in 6af9854, but breaks again after that commit was reverted. Does anyone have more context on why that commit was reverted?

**Comment by danaenz** [comment]
Hi all,

We've had this issue reported as well. Affected versions:
silverstripe/recipe-cms 1.1.3
 silverstripe/subsites 2.2.0
silverstripe/framework 4.1.3
silverstripe/cms 4.1.3

Since I need to fix this urgently for our client I'm attempting to find a workaround rather than roll back. I'm unsure which version I should roll back to if I need to, perhaps 1.1.0?

**Comment by ScopeyNZ** [comment]
> This appears to be fixed in [6af9854](https://github.com/silverstripe/silverstripe-subsites/commit/6af985420f896063694fc4f5aacfcc3c08c8a3a9), but breaks again after that commit was reverted. Does anyone have more context on why that commit was reverted?
> 
> Specifically, its breaking in the `Subsites.php` `accessible_sites` function, which returns 0 subsites even when "All subsites" access is selected.

Interesting. We identified subsites permission checking as being a bit flawed which @robbieaverill was kind enough to address in that PR. Unfortunately our test suites that test across with all modules installed went red implying it had some sort of integration flaws. We haven't had a chance to re-address this but I'll try to get this back on our radar.

**Comment by robbieaverill** [comment]
Yeah. That change introduced some strange regressions with the API behaviour that I modified so I reverted it until I have more time to look at it again

**Comment by adrexia** [comment]
@danaenz Looking at it, I'm not sure this has worked in any of the release branches for Silverstripe 4.  6af9854 was reverted before it hit a release. You could tie your composer to the sha, but then it risks hitting one of the reasons the suite tests failed. It seems like a pretty major issue though - the workaround is to give full CMS access to any user who needs to use tinymce (so not really workable).

**Comment by robbieaverill** [comment]
PR at https://github.com/silverstripe/silverstripe-subsites/pull/408 with a bit of a workaround in the meantime. We need to refactor the whole permission checking process in subsites, which was attempted recently and reverted, but until then I think this is a safe change which fixes this bug.

**Comment by robbieaverill** [comment]
This will be fixed in the next 2.2.x release, you can use `2.2.x-dev` in the meantime. Thanks for raising and contributing useful information everyone!
